keyboards/fancytech/fancyalice66/keymaps/via_encoder/keymap.c

```diff
@@ -0,0 +1,79 @@
+        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,             KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,          KC_UP,
+        KC_LCTL,          KC_LALT, KC_SPC,                    MO(1),   KC_SPC,                    KC_RALT,                   KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+
+    [1] = LAYOUT(
+        KC_TILD, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  RESET,            KC_PSCR, _______, _______,
+        _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, RGB_SAI, RGB_SAD, RGB_TOG, RGB_HUI,
+        _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______, _______,          RGB_HUD,
+        _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______,          RGB_VAI,
+	    _______,          _______, _______,                   _______, _______,                   RGB_MOD,                   _______, RGB_VAD, _______
+    ),
+
+    [2] = LAYOUT(
+        _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______,          _______,
+	    _______,          _______, _______,                   _______, _______,                   _______,                   _______, _______, _______
+    ),
+
+    [3] = LAYOUT(
+        _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______,          _______,
+	    _______,          _______, _______,                   _______, _______,                   _______,                   _______, _______, _______
+    )
+};
+
+#ifdef ENCODER_ENABLE
+bool encoder_update_user(uint8_t index, bool clockwise) {
+  if (index == 0) {
+    if (clockwise) {
+        tap_code(dynamic_keymap_get_keycode(biton32(layer_state), 4, 12));
```

Comment:
This isn't the right way to do this. This limits it to basic keycodes, only.

Ideally, VIA should update to support the encoder map feature, as that natively support the dynamic keymap feature, etc... but that is highly unlikely.

Reply:
I understand. Unfortunately Fancytech really wanted mapping support for VIA, that's the best I can do.